Document advanced binding options (#39839)
openshift/openshift-docs: applications/connecting_applications_to_services/binding-workloads-using-sbo.adoc

[id="binding-workloads-using-sbo"]

> = Binding workloads using Service Binding Operator

Reviewer: Doesn't a `workload` actually mean an `application`?

Reviewer: There seems to be a confusion in terms here

Author: @pmacik Yes, but when we are describing in general it was decided in our team SBO call meeting that we will use the term workload. And while explaining specific examples and uses cases we will use the term application.

[id="binding-workloads-using-sbo"]
= Binding workloads using Service Binding Operator
include::modules/common-attributes.adoc[]
include::modules/servicebinding-document-attributes.adoc[]
:context: binding-workloads-using-sbo

toc::[]

> Application developers must bind a workload to one or more backing services using a binding secret generated for the purpose of storing information to be consumed by the workload.

Reviewer 2: ```suggestion
Application developers must bind a workload to one or more backing services by using a binding secret. This secret is generated for the purpose of storing information to be consumed by the workload.
```
This sentence feels like there's a bit too much information in it. Consider creating two sentences for the content. Above is one suggestion for this edit.

> As an example, consider that the service you want to connect to is already exposing the binding data. You also need a workload to be used along with the `ServiceBinding` custom resource (CR). Through this `ServiceBinding` CR the workload sends a binding request with the details of the services to bind with.

Reviewer 2: ```suggestion
As an example, consider that the service you want to connect to is already exposing the binding data. In this case, you would also need a workload to be used along with the `ServiceBinding` custom resource (CR). Bu using this `ServiceBinding` CR, the workload sends a binding request with the details of the services to bind with.
```
- Use the conditional tense here for a hypothetical example.
- A couple of other small suggestions

.Example of `ServiceBinding` CR
----
apiVersion: binding.operators.coreos.com/v1alpha1
kind: ServiceBinding
metadata:
    name: spring-petclinic-rest
    namespace: my-postgresql
spec:
    services: <1>
    - group: postgres-operator.crunchydata.com
      version: v1beta1
      kind: PostgresCluster
      name: hippo
    - group: ""
      version: v1
      kind: Secret
      name: hippo-pguser-hippo
    application: <2>
      name: spring-petclinic-rest
      group: apps
      version: v1
      resource: deployments
----
<1> Specifies a list of service resources.

> <2> The sample application that points to a Deployment or any other similar resource with an embedded PodSpec.

Reviewer 2: ```suggestion
<2> The sample application that points to a Deployment or any other similar resource with an embedded PodSpec resource.
```

Author: It is just the embedded Podspec. I have removed the resource as that was a typo..

> resource.

Reviewer 2: ```suggestion
```

Author: removed this.
